src/Core/ServiceWrapper/Main.cs

```diff
@@ -605,6 +605,13 @@ public static void Run(string[] _args, ServiceDescriptor descriptor = null)
                     Registry.LocalMachine.OpenSubKey("System").OpenSubKey("CurrentControlSet").OpenSubKey("Services")
                         .OpenSubKey(d.Id, true).SetValue("Description", d.Description);
 
+                    if (d.StartMode == StartMode.Automatic && d.DelayedStart)
+                    {
+                        //TODO: replace by a better API after migrating to .NET 4
+                        Registry.LocalMachine.OpenSubKey("System").OpenSubKey("CurrentControlSet").OpenSubKey("Services")

```

Comment:
This is entirely the wrong way to do this. You should be using the SCM api for all of the service registry - hence the fact the SCM will not show the updated options until reboot.